### PR TITLE
feat(auto-salvage): v2 support

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#8430

#### What this PR does / why we need it:

Implement a new function to sort the keys of a map in ascending order.

#### Special notes for your reviewer:

`None`

#### Additional documentation or context

`None`

### CODE REVIEW

File: utils/misc.go [hunk 1]
```diff
@@ -119,3 +120,35 @@ func ConvertTypeToString[T any](value T) string {
...
+	if len(keys) == 0 {
+		return keys, nil
+	}
+
+	switch reflect.TypeOf(keys[0]).Kind() {
+	case reflect.String:
+		sort.Slice(keys, func(i, j int) bool {
+			return reflect.ValueOf(keys[i]).String() < reflect.ValueOf(keys[j]).String()
+		})
+	case reflect.Uint64:
```
Comment (james-munson) [inline utils/misc.go:145]: Not a huge deal, but why not make this library method work for any type that CanUint() ?

**Review by james-munson — APPROVED** [review]
Just one question, but LGTM.

**Review by shuo-wu — DISMISSED** [review]
LGTM

**Review by derekbit — APPROVED** [review]
LGTM.

@c3y1huang Ready to merge?

**Comment by c3y1huang** [comment]
> LGTM.
> 
> @c3y1huang Ready to merge?

Yes! It's just a general-purposed lib and is ready for merge. Thank you!

File: utils/misc.go (hunk 1 above)
Comment (c3y1huang) [reply utils/misc.go]: I am writing this for the v2 volume support auto salvage feature. We can extend with additional cases in future enhancements.

File: utils/misc.go [hunk 2]
```diff
@@ -119,3 +120,35 @@ func ConvertTypeToString[T any](value T) string {
 		return fmt.Sprintf("Unsupported type: %v", v.Kind())
 	}
 }
+
+// SortKeys sorts the keys of a map in ascending order.
+func SortKeys[K comparable, V any](mapObj map[K]V) ([]K, error) {
```
[the remark this answers is not in the document]
Comment (c3y1huang) [reply utils/misc.go]: Updated to make it more general.

**Review by derekbit — APPROVED** [review]
Thank you @c3y1huang

**Pull request merged**